File: src/plumbing_pub_sub/scripts/demo2_sub.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String
""""
        订阅流程
            1.导包
            2.初始化ros节点
            3.创建订阅者对象
            4.回调函数处理数据
            5.spin()
"""
def doMsg(msg):
    rospy.loginfo("我的订阅数据：%s",msg.data)

def lestener():
    #   2.初始化ros节点
    rospy.init_node("HuaHua")
    # 3.创建订阅者对象
    sub =rospy.Subscriber("fang",String,doMsg,queue_size=10)
    #         4.回调函数处理数据
    #         5.spin()
    rospy.spin()


if __name__ =="__main__":
    lestener()

File: src/plumbing_pub_sub/src/demo2_sub.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
/*
        订阅方实现：
            1.包含头文件
                ROS文本类型 --->std_msgs/String.h
            2.初始化ROS节点
            3.创建节点句柄
            4创建订阅者对象
            5.处理订阅到的数据
            6.声明一个spin()函数
*/
void doMsg(const std_msgs::String::ConstPtr &msg){
    // 通过msg获取并操作订阅后的数据
    
    ROS_INFO("翠花订阅的数据：%s",msg->data.c_str());
}

int main(int argc, char *argv[])
{
        setlocale(LC_ALL,"");
            // 2.初始化ROS节点
        ros::init(argc,argv,"cuihua");
            // 3.创建节点句柄
        ros::NodeHandle nh;
            // 4创建订阅者对象
        ros::Subscriber sub = nh.subscribe("fang",10,doMsg);
            // 5.处理订阅到的数据
        ros::spin();//回头，调用回调函数
    return 0;
}
